zeq/publisher.cpp

```diff
@@ -16,21 +17,55 @@
 #  include <zerobuf/Zerobuf.h>
 #endif
 
+#include <cstring>
 #include <map>
-#include <string.h>
 
 namespace zeq
 {
-namespace detail
+
+namespace
+{
+std::string _getApplicationName()
 {
-class Publisher : public Sender
+    // http://stackoverflow.com/questions/933850
+#ifdef _MSC_VER
+    char result[MAX_PATH];
+    const std::string execPath( result, GetModuleFileName( NULL, result,
+                                                           MAX_PATH ));
+#elif __APPLE__
+    char result[PATH_MAX+1];
+    uint32_t size = sizeof(result);
+    if( _NSGetExecutablePath( result, &size ) != 0 )
+        return std::string();
+    const std::string execPath( result );
+#else
+    char result[PATH_MAX];
+    const ssize_t count = readlink( "/proc/self/exe", result, PATH_MAX );
+    const std::string execPath( result, count > 0 ? count : 0 );
```

Comment:
Maybe /proc/self/exe doesn't exist, for example is someone tries in FreeBSD (also a UNIX flavour), and it will silently fail.